kbase-extension/static/kbase/js/kbaseNarrative.js

```diff
@@ -762,44 +868,50 @@ define([
             Jupyter.CellToolbar.global_show();
 
             if (Jupyter.notebook && Jupyter.notebook.metadata) {
-                var creatorId = Jupyter.notebook.metadata.creator || 'KBase User';
+                const creatorId = Jupyter.notebook.metadata.creator || 'KBase User';
                 DisplayUtil.displayRealName(creatorId, $('#kb-narr-creator'));
 
                 // This puts the cell menu in the right place.
-                $([Jupyter.events]).trigger('select.Cell', { cell: Jupyter.notebook.get_selected_cell() });
+                $([Jupyter.events]).trigger('select.Cell', {
+                    cell: Jupyter.notebook.get_selected_cell(),
+                });
             }
             if (this.getWorkspaceName() === null) {
-                KBFatal('Narrative.init', 'Unable to locate workspace name from the Narrative object!');
+                KBFatal(
+                    'Narrative.init',
+                    'Unable to locate workspace name from the Narrative object!'
+                );
                 this.loadingWidget.remove();
                 return;
             }
             this.initSharePanel();
             this.initStaticNarrativesPanel();
             this.updateDocumentVersion()
-                .then(function() {
-                    // init the controller
-                    return this.narrController.render();
-                }.bind(this))
-                .finally(function () {
-                    this.sidePanel.render();
-                }.bind(this));
-
-            $([Jupyter.events]).on('kernel_ready.Kernel',
-                (e) => {
-                    this.loadingWidget.updateProgress('kernel', true);
-                    this.jobCommChannel = new JobCommChannel();
-                    // TODO: This should be an event "kernel-ready", perhaps broadcast
-                    // on the default bus channel.
-                    this.jobCommChannel.initCommChannel()
-                        .then(() => this.loadingWidget.updateProgress('jobs', true))
-                        .catch((err) => {
-                            // TODO: put the narrative into a terminal state
-                            console.error('ERROR initializing kbase comm channel', err);
-                            KBFatal('Narrative.ini', 'KBase communication channel could not be initiated with the back end. TODO');
-                        });
-                }
-            );
-        }.bind(this));
+                .then(() => this.narrController.render())
+                .finally(() => this.sidePanel.render());
+        });
+        $([Jupyter.events]).on('kernel_connected.Kernel', () => {
```

Comment:
Here's the primary change - `kernel_ready.Kernel` only fires when the kernel is ready for input, and not busy. If there's something running, it'll never fire. This will look like there's a failure, when really the kernel is busily crunching some code.

`kernel_connected.Kernel` fires when the connection is set up properly. It should be much more consistent here.

Reply:
I also made the event response set up in parallel with the `notebook_loaded.Notebook` event above, instead of only set up when the Notebook event fires. I've never seen them not go in that order, but it's possible, and this is cleaner.